src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,180 @@
+
+            try
+            {
+                if (!IsDictionary(node))
+                {
+                    return DefaultVisit();
+                }
+
+                var initializerExtractionResults
+                    = node.Kind() switch
+                    {
+                        SyntaxKind.CollectionInitializerExpression
+                        => ExtractInitializersWithCollectionSyntax(node),
+                        SyntaxKind.ObjectInitializerExpression
+                        => ExtractInitializersWithObjectSyntax(node),
+                        _
+                        => throw new InvalidKindException {Kind = node.Kind()}
+                    };
+
+                if (!initializerExtractionResults.Success)
+                {
+                    return DefaultVisit();
+                }
+
+                var visitedInitializerSyntaxNodePairs = initializerExtractionResults.InitializerSyntaxNodePairs
+                    .Select(initializer => new KeyValuePair<SyntaxNode, SyntaxNode>(
+                        this.Visit(initializer.Key),
+                        this.Visit(initializer.Value))
+                    );
+
+                var replacementNode
+                    = BuildReplacementSyntaxWithCollectionInitialization(visitedInitializerSyntaxNodePairs);
+
+                if (replacementNode == default)
+                {
+                    var message
+                        = $"{nameof(NormalizeDictionaryInitialization)}: failed to find a {nameof(InitializerExpressionSyntax)} node in generated dictionary fragment";
+                    throw new NullReferenceException(message);
+                }
+
+                return base.VisitInitializerExpression(replacementNode as InitializerExpressionSyntax);
+            }
+            catch (InvalidKindException ike)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: {nameof(InitializerExpressionSyntax)} found with unexpected Kind {ike.Kind}");
+                throw;
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"{nameof(NormalizeDictionaryInitialization)}: unknown error");
+                throw;
+            }
+        }
+
+        private bool IsDictionary(InitializerExpressionSyntax initializerExpression)
+        {
+            var isDictionary = initializerExpression?.Parent?.ChildNodes()?.FirstOrDefault()?.GetText()?.ToString()
+                ?.StartsWith("Dictionary");
+            if (!isDictionary.HasValue)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: unable to retrieve the type information for {nameof(initializerExpression)}");
+                return false;
+            }
+
+            if (!isDictionary.Value)
+            {
+                return false;
+                // some other sort of object
+            }
+
+            return true;
+        }
```

Comment:
This is slightly more type-safe and specific. It also still passes all the tests:

```suggestion
        private static bool IsDictionary(InitializerExpressionSyntax initializerExpression) =>
            initializerExpression?.Parent is ObjectCreationExpressionSyntax objectCreationExpression &&
            objectCreationExpression.Type is GenericNameSyntax genericName &&
            genericName.Identifier.Text == "Dictionary";
```